CMakeLists.txt

```diff
@@ -14,12 +14,11 @@ endif ()
 option(FAASM_SELF_TRACING "Turn on system tracing using the logger" OFF)
 option(FAASM_PERF_PROFILING "Turn on profiling features as described in debugging.md" OFF)
 
-# SGX functionality
-# These options customise the SGX features _provided_ SGX is found
-option(FAASM_SGX "Enable SGX support" ON)
-option(FAASM_SGX_SIM_MODE "Turn on SGX sim mode" ON)
-option(FAASM_SGX_ATTESTATION "Turn on attestation" OFF)
-option(FAASM_SGX_XRA "Turn on support for XRA" OFF)
+# This option customises the SGX features _provided_ SGX is found:
+# - Disabled: disable all SGX features
+# - Simulation: use SGX in simulation mode (default)
+# - Hardware: use SGX in hardware mode
+option(FAASM_SGX_MODE "Enable SGX support" "Simulation")
```

Comment:
I think these terms are pretty self-explanatory but now the help text for the option isn't so clear, instead maybe:

```
option(FAASM_SGX_MODE "Type of SGX support: Disabled, Simulation or Hardware" "Simulation")
```